packages/account/src/BaseSmartAccount.ts

```diff
@@ -248,53 +248,52 @@ export abstract class BaseSmartAccount implements IBaseSmartAccount {
     this.validateUserOp(userOp, requiredFields);
 
     let finalUserOp = userOp;
-    const skipBundlerCall = skipBundlerGasEstimation ?? true;
+    const skipBundlerCall = skipBundlerGasEstimation ?? false;
     // Override gas values in userOp if provided in overrides params
     if (overrides) {
       userOp = { ...userOp, ...overrides };
     }
 
     Logger.log("userOp in estimation", userOp);
 
-    if (skipBundlerCall) {
-      if (this.paymaster && this.paymaster instanceof BiconomyPaymaster && paymasterServiceData?.mode === PaymasterMode.SPONSORED) {
-        if (!userOp.maxFeePerGas && !userOp.maxPriorityFeePerGas) {
-          throw new Error("maxFeePerGas and maxPriorityFeePerGas are required for skipBundlerCall mode");
+    if (this.paymaster && this.paymaster instanceof BiconomyPaymaster && paymasterServiceData?.mode === PaymasterMode.SPONSORED) {
+      if (!userOp.maxFeePerGas && !userOp.maxPriorityFeePerGas) {
+        throw new Error("maxFeePerGas and maxPriorityFeePerGas are required for skipBundlerCall mode");
+      }
+      // Making call to paymaster to get gas estimations for userOp
+      const { callGasLimit, verificationGasLimit, preVerificationGas, paymasterAndData } = await (
+        this.paymaster as IHybridPaymaster<SponsorUserOperationDto>
+      ).getPaymasterAndData(userOp, paymasterServiceData);
+      finalUserOp.verificationGasLimit = verificationGasLimit ?? userOp.verificationGasLimit;
+      finalUserOp.callGasLimit = callGasLimit ?? userOp.callGasLimit;
+      finalUserOp.preVerificationGas = preVerificationGas ?? userOp.preVerificationGas;
+      finalUserOp.paymasterAndData = paymasterAndData ?? userOp.paymasterAndData;
+    } else {
+      if (skipBundlerCall) {
+        Logger.warn("Skipped bundler call. If you are using paymasterAndData, generate data externally");
+        finalUserOp = await this.calculateUserOpGasValues(userOp);
+        finalUserOp.paymasterAndData = "0x";
+      } else {
+        if (!this.bundler) throw new Error("Bundler is not provided");
+        delete userOp.maxFeePerGas;
+        delete userOp.maxPriorityFeePerGas;
+        // Making call to bundler to get gas estimations for userOp
+        const { callGasLimit, verificationGasLimit, preVerificationGas, maxFeePerGas, maxPriorityFeePerGas } =
+          await this.bundler.estimateUserOpGas(userOp);
+        // if neither user sent gas fee nor the bundler, estimate gas from provider
+        if (!userOp.maxFeePerGas && !userOp.maxPriorityFeePerGas && (!maxFeePerGas || !maxPriorityFeePerGas)) {
+          const feeData = await this.provider.getFeeData();
```

Comment:
Would suggest to use from bundler, as provider one I have noticed does not match the bundler one on L2s
Bundler makes direct RPC call and does not use a library to get max fee values.
In the library somethings are hardcoded

Reply:
this is the fallback if bundler doesn't return